packages/kit/src/hooks/useFlowQueryRaw.ts

```diff
@@ -0,0 +1,50 @@
+import * as fcl from "@onflow/fcl"
+import {useQuery, UseQueryResult, UseQueryOptions} from "@tanstack/react-query"
+import {useCallback} from "react"
+import {useFlowQueryClient} from "../provider/FlowQueryClient"
+
+export interface UseFlowQueryRawArgs {
+  cadence: string
+  args?: (arg: typeof fcl.arg, t: typeof fcl.t) => unknown[]
+  query?: Omit<UseQueryOptions<unknown, Error>, "queryKey" | "queryFn">
+}
+
+/**
+ * useFlowQueryRaw
+ *
+ * Executes a Cadence script and returns the raw query result.
+ *
+ * @param params
+ *   - cadence: The Cadence script to run
+ *   - args: (optional) A function returning script arguments
+ *   - query: (optional) React Query settings (staleTime, retry, enabled, select, etc.)
+ * @returns {UseQueryResult<unknown, Error>}
+ */
+export function useFlowQueryRaw({
+  cadence,
+  args,
+  query: queryOptions = {},
+}: UseFlowQueryRawArgs): UseQueryResult<unknown, Error> {
+  const queryClient = useFlowQueryClient()
+
+  const fetchQueryRaw = useCallback(async () => {
+    if (!cadence) return null
+    return fcl.rawQuery({cadence, args})
+  }, [cadence, args])
+
+  // Encode the arguments to a JSON-CDC object so they can be deterministically
+  // serialized and used as the query key.
+  const encodedArgs = args?.(fcl.arg, fcl.t)?.map((x: any) =>
```

Comment:
This is in the `useFlowQuery` file as well. Can we move it somewhere its shared?